docs/4.3/aws_oss_guide.md

```diff
@@ -386,12 +379,308 @@ To upgrade to a newer version of Teleport:
+- [jq](https://stedolan.github.io/jq/) installed on you  local machine.
+- An AWS Account with Root Access
+
+### Create cluster role and role binding
+The first step is to create a cluster role and role binding that will allow the Teleport
+EC2 instance to impersonate other users, groups, and service accounts.
+
+The below command is ran on a machine with `kubectl` access to the cluster.
+
+```bash
+$ cat << 'EOF' | kubectl apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: teleport-impersonation
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - users
+  - groups
+  - serviceaccounts
+  verbs:
+  - impersonate
+- apiGroups:
+  - "authorization.k8s.io"
+  resources:
+  - selfsubjectaccessreviews
+  verbs:
+  - create
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: teleport-crb
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: teleport-impersonation
+subjects:
+- kind: Group
+  name: teleport-group
+- kind: User
+  name: system:anonymous
+EOF
+```
+If successful the terminal should output.
+```
+# clusterrole.rbac.authorization.k8s.io/teleport-impersonation created
+# clusterrolebinding.rbac.authorization.k8s.io/teleport-crb created
+```
+
+### Create IAM trust policy document
+This is the trust policy that allows the Teleport EC2 instance to assume a role.
+
+```bash
+$ cat > teleport_assume_role.json << 'EOF'
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+```
+This will create a `.json` file locally, the below command should then be ran to create the role.
+
+```bash
+$ ROLE_ARN=$(aws iam create-role --role-name teleport-roles --assume-role-policy-document file://teleport_assume_role.json | jq -r '.Role.Arn')
+```
+
+### Create IAM policy granting list-clusters and describe-cluster permissions (optional)
+
+This policy is necessary to create a `kubeconfig` file using the `aws eks update-kubeconfig`
+command. If you have another mechanism to create a kubeconfig file on the instance that runs
+Teleport, this step is not required.
+
+```bash
+cat > teleport_eks_desc_and_list.json << 'EOF'
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "eks:DescribeCluster",
+                "eks:ListClusters"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+POLICY_ARN=$(aws iam create-policy --policy-name teleport-policy --policy-document file://teleport_eks_desc_and_list.json | jq -r '.Policy.Arn')
+aws iam attach-role-policy --role-name teleport-role --policy-arn $POLICY_ARN
+```
+
+### Update aws-auth configmap
```

Comment:
nit: "Map IAM role to Kubernetes group"
configmap is just an implementation detail and most readers won't know about it